client/python/cryoet_data_portal/src/cryoet_data_portal/_gql_base.py

```diff
@@ -14,6 +14,15 @@ def __init__(self, field: "GQLField", operator, value: Optional[Any]):
 
     def to_gql(self) -> Dict[str, Any]:
         fieldname = self.field.to_gql()
+        if self.field.is_relationship():
+            related_cls = self.field.get_related_class()
+            scalar_fields = related_cls._get_scalar_fields()
+            if isinstance(fieldname, str):
+                fieldname = [fieldname]
+
+            raise Exception(
+                f"\"{'.'.join(fieldname)}\" is an object and can't be filtered directly. Please filter on one of its scalar attributes instead: {', '.join(scalar_fields)}",
```

Comment:
Do we want to say "filtered directly" here or "compared directly"?

Reply:
I updated the message!